src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/security/authorization/OrderlyWebAuthorizationGenerator.kt

```diff
@@ -0,0 +1,99 @@
+package org.vaccineimpact.orderlyweb.security.authorization
+
+import org.jooq.Record
+import org.pac4j.core.authorization.generator.AuthorizationGenerator
+import org.pac4j.core.context.WebContext
+import org.pac4j.core.profile.CommonProfile
+import org.vaccineimpact.orderlyweb.db.JooqContext
+import org.vaccineimpact.orderlyweb.db.Tables.*
+import org.vaccineimpact.orderlyweb.db.asTemporaryTable
+import org.vaccineimpact.orderlyweb.db.fromJoinPath
+import org.vaccineimpact.orderlyweb.db.withTemporaryTable
+import org.vaccineimpact.orderlyweb.models.Scope
+import org.vaccineimpact.orderlyweb.models.permissions.PermissionSet
+import org.vaccineimpact.orderlyweb.models.permissions.ReifiedPermission
+
+class OrderlyWebAuthorizationGenerator : AuthorizationGenerator<CommonProfile>
```

Comment:
'Generator' rather than 'repo' or 'provider'? It's fetching things rather than creating anything. I suppose we're kind of stuck with it since it derives from pac4j though!

Reply:
I've gone with `AuthorizationRepository` to make this clearer